app/models/docker_builder_service.rb

```diff
@@ -0,0 +1,93 @@
+require 'docker'
+
+class DockerBuilderService
+  attr_reader :build, :image
+
+  def initialize(build)
+    @build = build
+  end
+
+  def build!(image_name: nil, push: true)
+    # TODO: check if there's already a docker build
+
+    job = build.create_docker_build_job do |job|
+      job.project = build.project
+      job.user_id = build.created_by || NullUser.new.id
+      job.command = ''    # doesn't seem right, but command can't be nil
+      job.commit = build.git_sha
+      job.tag = build.git_ref
+    end
+
+    build.save!
+
+    job_execution = JobExecution.start_job(build.git_sha, job) do |execution, tmp_dir|
```

Comment:
its [implicit return](https://github.com/zendesk/samson/pull/453/files#diff-199416dcde8818efe1b2180a2a2e9141L176) is used by the subscribe call a few lines down from here. The `tap` makes it look like it was originally supposed to return an instance.